src/index.ts

```diff
@@ -0,0 +1,88 @@
+import crypto from 'crypto';
+import { Request, RequestHandler, NextFunction } from 'express';
+
+import stringToBuffer from './stringToBuffer';
+import validateArguments from './validateArguments';
+import AuthError from './errors';
+
+export type Options = {
+    algorithm: string;
+    identifier: string;
+    header: string;
+    maxInterval: number;
+    minInterval: number;
+}
+
+const defaults: Options = {
+    algorithm: 'sha256',
+    identifier: 'HMAC',
+    header: 'authentication',
+    maxInterval: 60 * 5,
+    minInterval: 0 
+}
+
+export { AuthError };
+
+export function HMAC(secret: string, options: Partial<Options> = {}): RequestHandler {
+    const mergedOpts: Options = { ...defaults, ...options };
+
+    const error: TypeError | void = validateArguments(secret, mergedOpts);
```

Comment:
Curious what the reasoning was behind returning the errors from the `validateArguments` function rather than throwing them there and catching them here?

Returning exceptions rather than throwing allows those exceptions to be mistakenly ignored by other developers, whereas if you throw you're forced to deal with it in some way(try/catch). At the moment `validateArguments` only ever returns `TypeError` if something is broken so you're always throwing the error on the next line anyway.